scripts/apps/authoring/authoring/directives/AuthoringDirective.js

```diff
@@ -599,8 +614,9 @@ export function AuthoringDirective(superdesk, superdeskFlags, authoringWorkspace
                     return;
                 }
 
-                // Check if there's unpublished related items
-                const related = relationsService.getRelatedItemsWithoutMediaGallery($scope.item, $scope.fields);
+                // Check if there's unpublished related items without media-gallery
+                const related = relationsService.getRelatedItemsWithoutMediaGallery($scope.updatedItem || $scope.item,
```

Comment:
You are introducing another scope variable here - `updatedItem`. It's now clear to me what are you doing, but it would be hard to understand it for someone who hasn't looked at this PR.

Please move all related item related logic to `RelatedItemsDirective.ts`. Let's discuss it if you see any issues with that.  I don't see any reasons why this couldn't be done. You have IDs on the item, you pass it to directive which renders the items, and it also takes care of fetching the items itself without any changes to `AuthoringDirective.js` required.

Reply:
Ok, I will check it out. and let you know if I need something.